pkg/gits/credentialhelper/git_credential.go

```diff
@@ -0,0 +1,122 @@
+package credentialhelper
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+)
+
+// GitCredential represents the different parts of a git credential URL
+// See also https://git-scm.com/docs/git-credential
+type GitCredential struct {
+	Protocol string
+	Host     string
+	Path     string
+	Username string
+	Password string
+}
+
+// CreateGitCredential creates a CreateGitCredential instance from a slice of strings where each element is a key/value pair
+// separated by '='.
+func CreateGitCredential(lines []string) (GitCredential, error) {
+	var credential GitCredential
+
+	if lines == nil {
+		return credential, errors.New("no data lines provided")
+	}
+
+	fieldMap, err := util.ExtractKeyValuePairs(lines, "=")
+	if err != nil {
+		return credential, errors.Wrap(err, "unable to extract git credential parameters")
+	}
+	for key, value := range fieldMap {
+		v := reflect.ValueOf(&credential).Elem().FieldByName(strings.Title(key))
```

Comment:
> Why do you need this trick? I am not a bit fan of reflect in production code even though makes the code more generic.

I don't need it :-) I think I just did not want to type things out. As you say, it's a bit more generic. Any concrete reason why you don't like reflection? To be fair, I was undecided on whether I should type things our or whether I should use reflection. Coming from the Java world I am quite used to using reflection, but it might not be as common in Go. 

If you feel strongly about it, I can change it

Reply:
> I think that you can also use json.Marshal(fieldMap) and then json.Unmarshal(&GitCredentials). It is a bit more portable if the types of the struct gets changed.

that's probably one way, even though this under the hood probably does the same type of reflection as well. Really, if it comes down to it, you can just type it out. 

To be fair, people might be more familiar with the Marshalling technique.

Reply:
Actually there is a difference, your marshalling technique will silently ignore unknown fields.

Reply:
Using the JSON technique since it is more familiar